src/com/magento/idea/magento2plugin/actions/generation/NewWebApiDeclarationAction.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+package com.magento.idea.magento2plugin.actions.generation;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiDirectory;
+import com.jetbrains.php.lang.documentation.phpdoc.psi.PhpDocComment;
+import com.jetbrains.php.lang.psi.elements.Method;
+import com.jetbrains.php.lang.psi.elements.PhpClass;
+import com.magento.idea.magento2plugin.MagentoIcons;
+import com.magento.idea.magento2plugin.actions.generation.dialog.NewWebApiDeclarationDialog;
+import com.magento.idea.magento2plugin.project.Settings;
+import com.magento.idea.magento2plugin.util.php.PhpPsiElementsUtil;
+import org.jetbrains.annotations.NotNull;
+
+public class NewWebApiDeclarationAction extends AnAction {
+
+    public static final String ACTION_NAME = "Create a new WebApi declaration for this method";
+    public static final String ACTION_DESCRIPTION = "Create a new Magento 2 WebApi XML declaration";
+    private Method currentPhpMethod;
+
+    /**
+     * New WebApi declaration action constructor.
+     */
+    public NewWebApiDeclarationAction() {
+        super(ACTION_NAME, ACTION_DESCRIPTION, MagentoIcons.MODULE);
+    }
+
+    @Override
+    public void update(final @NotNull AnActionEvent event) {
+        setIsAvailableForEvent(event, false);
+        final Project project = event.getProject();
+        final Method method = PhpPsiElementsUtil.getPhpMethod(event);
+
+        if (project == null || !Settings.isEnabled(project) || method == null) {
+            return;
+        }
+
+        if (method.getContainingClass() == null) {
+            return;
+        }
+
+        final PhpDocComment classDocComment = method.getContainingClass().getDocComment();
+
+        if (!method.getAccess().isPublic()
+                || method.getName().equals("__construct")
```

Comment:
We have the same string literal here: `com.magento.idea.magento2plugin.magento.files.Plugin#CONSTRUCT_METHOD_NAME`
Could you please put them in some unified place?

Reply:
Done.